src/Form/MailTemplateCustomDataType.php

```diff
@@ -28,13 +29,23 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
         if (null !== $options['template']) {
             $customFields = $this->mailTemplateHelper->getCustomFields($options['template']);
 
-            foreach ($customFields as $key => $label) {
-                $builder->add($key, TextType::class, [
-                    'label' => $label,
-                    'mapped' => false,
-                    'data' => isset($options['data'][$key]) && !empty($options['data'][$key]) ? $options['data'][$key] : '',
-                    'required' => false,
-                ]);
+            foreach ($customFields as $key => $value) {
+                if ('textArea' === $value['type']) {
```

Comment:
Consider adding a `const`, `TYPE_TEXTAREA = 'textarea'`, say, in this class and use it in stead of writing `'textArea`' in multiple places.

Reply:
Added in 81896df1e4a3900c171dacc9dbf863168bf41171.